framework/src/main/java/org/checkerframework/framework/util/dependenttypes/DependentTypesHelper.java

```diff
@@ -195,9 +196,38 @@ private void viewpointAdaptExecutable(
             receiver = FlowExpressions.internalReprOf(factory, receiverTree);
         }
 
-        List<FlowExpressions.Receiver> argReceivers = new ArrayList<>(args.size());
-        for (ExpressionTree argTree : args) {
-            argReceivers.add(FlowExpressions.internalReprOf(factory, argTree));
+        List<Receiver> argReceivers = new ArrayList<>();
+        boolean varargs = false;
+        if (tree.getKind() == Kind.METHOD_INVOCATION) {
+            ExecutableElement methodCalled = TreeUtils.elementFromUse((MethodInvocationTree) tree);
+            if (methodCalled != null && methodCalled.isVarArgs()) {
+                varargs = true;
+                for (int i = 0; i < methodCalled.getParameters().size() - 1; i++) {
+                    argReceivers.add(FlowExpressions.internalReprOf(factory, args.get(i)));
+                }
+                List<Receiver> initializers = new ArrayList<>();
+                for (int i = methodCalled.getParameters().size() - 1; i < args.size(); i++) {
+                    initializers.add(FlowExpressions.internalReprOf(factory, args.get(i)));
+                }
+                Element varargsElement =
+                        methodCalled.getParameters().get(methodCalled.getParameters().size() - 1);
+                TypeMirror tm = ElementUtils.getType(varargsElement);
+                if (initializers.size() == 1
+                        && initializers.get(0).getType().toString().equals(tm.toString())) {
```

Comment:
See org.checkerframework.framework.util.AnnotatedTypes#expandVarArgs:
```
        if (parameters.size() == args.size()) {
            // Check if one sent an element or an array
            AnnotatedTypeMirror lastArg = atypeFactory.getAnnotatedType(args.get(args.size() - 1));
            if (lastArg.getKind() == TypeKind.ARRAY
                    && getArrayDepth(varargs) == getArrayDepth((AnnotatedArrayType) lastArg)) {
                return parameters;
            }
        }
```

You should do something like that here.